src/lighteval/models/base_model.py

```diff
@@ -322,6 +324,74 @@ def greedy_until_with_logits(
             override_bs=override_bs,
         )
 
+    def greedy_until_multi_turn(self, requests: list[GreedyUntilMultiTurnRequest], override_bs: Optional[int] = None) -> GenerateMultiTurnReturn:
+        for request in requests:
+            request.stop_sequence = as_list(request.stop_sequence) + [self.tokenizer.eos_token]
+            request.tokenized_context = self.tok_encode(request.context)
+
+        dataset = GenerativeTaskDataset(requests=requests, dataset_splits=self.DATASET_SPLITS)
+        dataloader = DataLoader(dataset, batch_size=1, collate_fn=lambda batch: batch)
+
+        results = []
+
+        if self.accelerator:
+            dataloader = self.accelerator.prepare(dataloader)
+
+        # Always batch size 1 for multi-turn
+        for batch in tqdm(
+            dataloader, desc="Greedy Multi Turn generation", position=1, leave=False, disable=self.disable_tqdm
+        ):
+            # NOTE: we are assuming all items in a batch behave similarly (same
+            # stop_tokens and max_tokens genrated) which is not necessarily
+            # the case! Because of that we only use batch size of 1
+            stop_tokens = batch[0].stop_sequence
+            max_generated_tokens = batch[0].generation_size
+            contexts = [c.context for c in batch]
+            max_context_size_allowed = self.max_length - max_generated_tokens
+
+            multi_turn_context = "" # contexts[0][0]
+            model_answers = []
+            for i, context in enumerate(contexts[0]):
```

Comment:
Does this take into account the chat templating?

Reply:
There is not chat templating for now, it's a work in progress. It works but i'm still trying to find a way to add it in cleanly. Especially managing few shots and templating.

Reply:
That works. I think we'll have to move the prompt management (few shot, CoT, chat templating, ...) in its own class at one point btw (everything is in the few shot manager atm)